sdks/go/pkg/beam/core/funcx/sideinput_test.go

```diff
@@ -39,6 +39,7 @@ func TestIsIter(t *testing.T) {
 		{func(*typex.EventTime, *int, *string) bool { return false }, false},            // EventTimes disallowed
 		{func(*int, *typex.Y, *typex.Z) bool { return false }, false},                   // too many values
 		{func(*typex.EventTime, *int, *typex.Y, *typex.Z) bool { return false }, false}, // too many values, EventTimes disallowed
+		{func(any) bool { return false }, false},                                        // any is not allowed as a param
```

Comment:
Nit, this needs to be *any since these iterator functions always take the pointer to the value, not the value itself.

Reply:
😫So sorry for that.
I have fixed it.
Another question is: can 'emit function' take both pointer and non-pointer as params? If it can, I will add test cases for *any and any in output_test.go.

Reply:
Great question and point!

Yes emitter functions can take pointer elements. And similarly, since the iterator function parameter is the pointer to an element type, you can pass in a pointer to a pointer. 

Those would be `func(*any)` and `func(**any) bool`

Pointers to pointers are generally rare in Go, but it's consistent with the intended handling. If Go develops a standard approach to iterators, we'd also support that more directly, and hopefully avoid the ** question.

Reply:
Thanks!
I made add one line of code for *any in the emitter.
I think we can make future changes if Go could develop a standard approach to iterators.